ulauncher/modes/extensions/extension_registry.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import logging
+from typing import Iterator
+
+from ulauncher.modes.extensions import extension_finder
+from ulauncher.modes.extensions.extension_controller import (
+    ExtensionController,
+    ExtensionNotFoundError,
+    lifecycle_events,
+)
+
+logger = logging.getLogger()
+_ext_controllers: dict[str, ExtensionController] = {}
+
+
+def get(ext_id: str) -> ExtensionController:
+    """Get an extension from the registry or raise if not found."""
+    controller = _ext_controllers.get(ext_id)
+    if not controller:
+        msg = f"Extension with ID '{ext_id}' not found in registry"
+        raise ExtensionNotFoundError(msg)
+
+    return controller
+
+
+def iterate() -> Iterator[ExtensionController]:
+    """Iterate over registered extension controllers."""
+    yield from _ext_controllers.values()
+
+
+def load(ext_id: str, path: str | None = None) -> ExtensionController:
+    """Load an extension controller into the registry without starting it."""
+    if not path:
+        path = extension_finder.locate(ext_id)
+        if not path:
+            _ext_controllers.pop(ext_id, None)
```

Comment:
I think this is a strange non-explicit side effect. If you call load it can do the opposite. Not that I haven't introduced side effects myself, but it should at least have a comment about why I think.

Reply:
> It does, however, trigger reload dbus command, which would crash trying to load a deleted extension.

That makes sense :+1: I think if you add this as a comment so we understand this line in 6 months time when reading it, that's fine too.